### PR TITLE
Replace checked-in ml-commons dependency for 2.0

### PR DESCRIPTION
**Pull request by jackiehanyang**
### Description
Replace checked-in ml-commons dependency for 2.0, using alpha remote file for now
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/495
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: doctest/build.gradle [hunk 1]
```diff
@@ -48,6 +48,9 @@ doctest.finalizedBy stopOpenSearch
 build.dependsOn doctest
 clean.dependsOn(cleanBootstrap)
 
+String mlCommonsRemoteFile = 'https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.0.0-alpha1/latest/linux/x64/builds/opensearch/plugins/opensearch-ml-2.0.0.0-alpha1.zip'
```
Comment (joshuali925) [inline doctest/build.gradle:51]: sql is on rc1, use this instead? https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.0.0-rc1/2500/linux/x64/tar/builds/opensearch/plugins/opensearch-ml-2.0.0.0-rc1.zip

also is it possible to substitute version with variable?

Comment (dblock) [reply doctest/build.gradle:51]: something like https://github.com/opensearch-project/cross-cluster-replication/pull/366

Comment (jackiehanyang) [reply doctest/build.gradle]: updated it in the latest revision

**Review by joshuali925 — APPROVED** [review]

File: doctest/build.gradle (hunk 1 above)
Comment (jackiehanyang) [reply doctest/build.gradle]: Will update it to remove zip dependency

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Pull request merged**